feature_engine/encoding/decision_tree.py

```diff
@@ -291,6 +342,15 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         """
         X = self._check_transform_input_and_state(X)
         _check_contains_na(X, self.variables_)
+        # replace unseen values if unseen==encode
+        if self.unseen == "encode":
+            for column, col_values in self._categories.items():
+                X[column] = np.where(X[column].isin(col_values) | X[column].isna(),
+                                     X[column], self.fill_value)
+        elif self.unseen == "ignore":
+            for column, col_values in self._categories.items():
+                X[column] = np.where(X[column].isin(col_values) | X[column].isna(),
+                                     X[column], np.nan)
```

Comment:
I am not sure this logic works as intended. First we replace the unseen with either a number or nan. If it is nan, then the transform step below should fail, because sklearn decision trees do not take nan as input. And if it replaces nan with a number, then the tree is going to use that number as input and return a prediction at the back of that. 

Somehow, we should first obtain the predictions over the permitted values and then replaced unseen by either nan or the fill_value.

Reply:
Hey @solegalli ! 
I implemented the changes you suggested. 
About the np.nan or fill_value replacement I used a workaround in the transform step:
* first I get a mask of those unknown
* I convert them to a known value to make succeed the encoding 
* in the end I replace the values in the mask with the fill_value or np.nan

Another point I would like to ask is that I am defining the `unseen` parameter of the OrdinalEncoder as `self.unseen`, but with the current developments I think I should keep it to `"raise"` because in any case it will receive seen categories. 
Here the code I am referring to. 
```
cat_encoder = OrdinalEncoder(
            encoding_method=self.encoding_method,
            variables=variables_,
            missing_values="raise",
            ignore_format=self.ignore_format,
            unseen="raise", # unseen=self.unseen,
        )
```